--- a/CMakeLists.txt.orig	2015-02-19 12:59:55.166490726 +0100
+++ b/CMakeLists.txt	2015-02-19 13:00:33.616492645 +0100
@@ -1270,7 +1270,6 @@
 endif()
 
 # TinyXML
-add_subdirectory(thirdparty/tinyxml)
 target_link_libraries(openclonk tinyxml)
 
 if(WIN32)
--- a/src/lib/StdMeshLoaderXml.cpp.orig	2015-02-19 13:39:47.197610077 +0100
+++ b/src/lib/StdMeshLoaderXml.cpp	2015-02-19 13:40:05.244610977 +0100
@@ -19,7 +19,7 @@
 #include "C4Include.h"
 #include "StdMesh.h"
 #include "StdMeshLoader.h"
-#include <tinyxml/tinyxml.h>
+#include "tinyxml.h"
 
 // Helper class to load things from an XML file with error checking
 class StdMeshLoader::StdMeshXML
